identity/handler.go

```diff
@@ -1030,16 +1030,28 @@ func (h *Handler) deleteIdentityCredentials(w http.ResponseWriter, r *http.Reque
 			h.r.Writer().WriteError(w, r, err)
 			return
 		}
-	case CredentialsTypePassword, CredentialsTypeCodeAuth:
-		h.r.Writer().WriteError(w, r, errors.WithStack(herodot.ErrBadRequest.WithReasonf("You cannot remove first factor credentials.")))
-		return
-	case CredentialsTypeOIDC:
-		if err := identity.deleteCredentialOIDCFromIdentity(r.URL.Query().Get("identifier")); err != nil {
+	case CredentialsTypePassword, CredentialsTypeCodeAuth, CredentialsTypeOIDC:
+		firstFactor, err := h.r.IdentityManager().CountActiveFirstFactorCredentials(r.Context(), identity)
+		if err != nil {
 			h.r.Writer().WriteError(w, r, err)
 			return
 		}
+		if firstFactor < 2 {
+			h.r.Writer().WriteError(w, r, errors.WithStack(herodot.ErrBadRequest.WithReasonf("You cannot remove the last first factor credential.")))
+			return
+		}
+		switch cred.Type {
+		case CredentialsTypePassword, CredentialsTypeCodeAuth:
+			identity.DeleteCredentialsType(cred.Type)
```

Comment:
If an identity as both an `oidc` credential with an identifier derived from the subject, and a password credential with an identifier set to the email address, and we delete the password credential, then the email address will not be a credential identifier any more, right? Is that what we want?

I would have expected that we just clear the password credentials config, but keep the identifier in place. Maybe `identity.Validate()` would do exactly that, but in any case we should add a test for it. WDYT?